creator-node/src/snapbackSM/syncHistoryAggregator.js

```diff
@@ -0,0 +1,125 @@
+})
+
+// Make key expire in 7 days in seconds
+const EXPIRATION = 7 /* days */ * 24 /* hr */ * 60 /* min */ * 60 /* s */
+
+/**
+ * Class intended to:
+ * - Record the number of the current day's successful and failed sync attempts
+ * - Record the timestamp of the most recent successful and failed sync
+ */
+
+// Note: When recording a sync 'success' or 'fail', it implies that the secondary was able to respectfully
+// successfully or unsuccessfully take the primary's exported data, and update its own state
+
+class SyncHistoryAggregator {
+  static async recordSyncSuccess (logContext) {
+    await SyncHistoryAggregator.recordSyncData({
+      state: SYNC_STATES.success,
+      timeOfEvent: moment().format('MM-DD-YYYYTHH:MM:SS:sss'),
+      logContext
+    })
+  }
+
+  static async recordSyncFail (logContext) {
+    await SyncHistoryAggregator.recordSyncData({
+      state: SYNC_STATES.fail,
+      timeOfEvent: moment().format('MM-DD-YYYYTHH:MM:SS:sss'),
+      logContext
+    })
+  }
+
+  static async recordSyncData ({ state, timeOfEvent, logContext }) {
+    const logger = genericLogger.child(logContext)
+
+    try {
+      // Update aggregate sync data
+      const aggregateSyncKeys = SyncHistoryAggregator.getAggregateSyncKeys()
+
+      // Get the TTL if the key exists for the number of `state` syncs, and update the value by 1
+      const aggregateSyncKeyStateTTL = await this.getKeyTTL(aggregateSyncKeys[state])
+      await redisClient.incr(aggregateSyncKeys[state])
+      await redisClient.expire(aggregateSyncKeys[state], aggregateSyncKeyStateTTL)
+
+      // Update latest sync data
+      const latestSyncKeys = SyncHistoryAggregator.getLatestSyncKeys()
+
+      const latestSyncKeyTTL = await this.getKeyTTL(latestSyncKeys[state])
+      await redisClient.set(latestSyncKeys[state], timeOfEvent, 'EX', latestSyncKeyTTL)
+
+      logger.info(`SyncHistoryAggregator - Successfully tracked ${state} sync at ${timeOfEvent}`)
+    } catch (e) {
+      // Only log error to not block any main thread
+      logger.error(`SyncHistoryAggregator - Failed to track ${state} sync at ${timeOfEvent}: ${e.toString()}`)
+    }
+  }
+
+  static async getKeyTTL (key) {
+    const ttl = await redisClient.ttl(key)
+    return ttl && ttl > 0 ? ttl : EXPIRATION
+  }
+
+  static async getAggregateSyncData () {
+    const { success, fail } = SyncHistoryAggregator.getAggregateSyncKeys()
+
+    let successCount = await redisClient.get(success)
+    let failCount = await redisClient.get(fail)
+
+    successCount = successCount ? parseInt(successCount) : 0
+    failCount = failCount ? parseInt(failCount) : 0
+
+    const currentAggregateData = {
+      success: successCount,
+      fail: failCount,
+      triggered: successCount + failCount
+    }
+
+    // Structure: {triggered: <number>, success: <number>, fail: <number>}
+    return currentAggregateData
+  }
+
+  static async getLatestSyncData () {
+    const { success, fail } = SyncHistoryAggregator.getLatestSyncKeys()
```

Comment:
variable naming of `success` and `fail` is def confusing, maybe something like `latestSyncSuccessRedisKey` or `latestSyncSuccessKey` etc would be better

Reply:
I thought it would be clear with the given method name called `getLatestSyncData()` that the inner variable names are relevant to the latest sync data. I think it's a bit excessive to give these keys the longer variable names you suggested.

Reply:
the function name doesn't matter bc it returns an object, so the code interacting with the resonse looks like this: `await redisClient.get(success)`. from this line it's not clear at all what `success` signifies

anyway, nbd can leave if you want